cmake/zephyr/Kconfig

```diff
@@ -7,3 +7,9 @@ config TOOLCHAIN_ZEPHYR_0_16
 config TOOLCHAIN_ZEPHYR_SUPPORTS_THREAD_LOCAL_STORAGE
 	def_bool y
 	select TOOLCHAIN_SUPPORTS_THREAD_LOCAL_STORAGE
+
+config NEWLIB_SUPPORTED
```

Comment:
I don't think a generic option should be in here.  For example how will the arm-gnu-embedded toolchain enable `NEWLIB_SUPPORTED`.  This should be in Zephyr proper and have something like:

```
config TOOLCHAIN_ZEPHYR_SUPPORTS_NEWLIB`
  def_bool y
  select NEWLIB_SUPPORTED
```

Reply:
or add `select NEWLIB_SUPPORTED` under `config TOOLCHAIN_ZEPHYR_0_16`

Reply:
I've already got a proposed patch which does that (no need for the version check, Zephyr SDK has always supported newlib). I was just following the suggestion about where similar mechanism for Picolibc should live, and that will live in sdk-ng. Happy to have it in Zephyr instead; it's a lot easier to manage there anyways.